Build error on Linux: libbinaryen.a

Getting the following error when I run `cmake --build`:

```
gmake[2]: *** No rule to make target `deps/lib/libbinaryen.a', needed by `src/libhera.so'.  Stop.
gmake[1]: *** [src/CMakeFiles/hera.dir/all] Error 2
gmake: *** [all] Error 2
```

Manually copying `libbinaryen.a` from `deps/lib64` seems to have worked.

Hera commit is https://github.com/ewasm/hera/commits/753dafc4f6e26d47fbbf3a119321689ca785b1b1. I'm running vanilla Amazon Linux:

```
> uname -a
Linux ip-172-31-69-70.ec2.internal 4.14.62-70.117.amzn2.x86_64 #1 SMP Fri Aug 10 20:14:53 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
> cmake --version
cmake version 3.12.2
```

Note that the same error (and workaround) was reported here: https://github.com/moneroexamples/compile-cpp-ethereum-on-arch/issues/3

note that this also sometimes happens on aleth build with `libbinaryen.a`
and `libmpir.a`. I am guessing that binaryen's cmake package may have an
issue.

On Tue, Sep 25, 2018, 16:38 Lane Rettig <notifications@github.com> wrote:

> Getting the following error when I run cmake --build:
>
> gmake[2]: *** No rule to make target `deps/lib/libbinaryen.a', needed by `src/libhera.so'.  Stop.
> gmake[1]: *** [src/CMakeFiles/hera.dir/all] Error 2
> gmake: *** [all] Error 2
>
> Manually copying libbinaryen.a from deps/lib64 seems to have worked.
>
> Hera commit is
> https://github.com/ewasm/hera/commits/753dafc4f6e26d47fbbf3a119321689ca785b1b1.
> I'm running vanilla Amazon Linux:
>
> > uname -a
> Linux ip-172-31-69-70.ec2.internal 4.14.62-70.117.amzn2.x86_64 #1 SMP Fri Aug 10 20:14:53 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
> > cmake --version
> cmake version 3.12.2
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ewasm/hera/issues/431>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHEQh6q8XP8RD4iw3EZ1etfySfHDkfkqks5uepRNgaJpZM4W5b5z>
> .
>

--- a/cmake/ProjectBinaryen.cmake
+++ b/cmake/ProjectBinaryen.cmake
@@ -4,7 +4,6 @@ endif()
 set(ProjectBinaryenIncluded TRUE)
 
 include(ExternalProject)
-include(GNUInstallDirs)
 
 if(MSVC)
     # Overwrite build and install commands to force Release build on MSVC.
@@ -24,7 +23,7 @@ set(source_dir ${prefix}/src/binaryen)
 set(binary_dir ${prefix}/src/binaryen-build)
 # Use source dir because binaryen only installs single header with C API.
 set(binaryen_include_dir ${source_dir}/src)
-set(binaryen_library ${prefix}/${CMAKE_INSTALL_LIBDIR}/${CMAKE_STATIC_LIBRARY_PREFIX}binaryen${CMAKE_STATIC_LIBRARY_SUFFIX})
+set(binaryen_library ${prefix}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}binaryen${CMAKE_STATIC_LIBRARY_SUFFIX})
 # Include also other static libs needed:
 set(binaryen_other_libraries
     ${binary_dir}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}wasm${CMAKE_STATIC_LIBRARY_SUFFIX}
@@ -46,6 +45,7 @@ ExternalProject_Add(binaryen
     URL_HASH SHA256=19439e41dc576446eaae0c4a8e07d4cd4c40aea7dfb0a6475b925686852f8006
     CMAKE_ARGS
     -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+    -DCMAKE_INSTALL_LIBDIR=lib
     -DCMAKE_BUILD_TYPE=Release
     -DBUILD_STATIC_LIB=ON
     ${build_command}

